public/video-ui/src/components/FormFields/TextInputTagPicker.js

```diff
@@ -40,33 +38,30 @@ export default class TextInputTagPicker extends React.Component {
   }
 
   selectNewTag = (newFieldValue) => {
-
       this.setState({
         inputString: ''
       });
 
       this.props.onUpdate(newFieldValue);
-  }
+  };
 
   getYoutubeInputValue = () => {
+    const existingValueSet = new Set(this.props.tagValue.map(_ => _.id));
+    const newValueSet = new Set();
+
     return this.state.inputString.split(',')
-      .filter(keyword => keyword.length !== 0)
-      .map(keyword => keyword.trim())
-      .reduce((keywordsAsObjects, keyword) => {
-        if (
-          this.props.tagValue.every(value => {
-            return value.id !== keyword;
-          })
-        ) {
-          keywordsAsObjects.push(
-            {
-              id: keyword,
-              webTitle: keyword
-            }
-          );
+      .filter(candidate => candidate.length !== 0)
+      .map(candidate => candidate.trim())
+      .reduce((acc, candidate) => {
+        const lowerCaseCandidate = candidate.toLowerCase();
+        const isNewValue = !existingValueSet.has(lowerCaseCandidate) && !newValueSet.has(lowerCaseCandidate);
+
+        if (isNewValue) {
+          acc.push({ id: candidate, webTitle: candidate});
         }
 
-        return keywordsAsObjects;
+        newValueSet.add(candidate);
```

Comment:
What is `newValueSet` being used for and why are we updating it from inside the reduce?

Reply:
`newValueSet` represents the new tags being added to the field. This field accepts a comma separated string, so we add to `newValueSet` to ensure input of `Foo, bar, baz, foo` results in `Foo, bar, baz` as `newValueSet` will be `Set('foo', 'bar', 'baz')` and the second `foo` is ignored. Its not especially obvious, I agree.